client.go

```diff
@@ -26,11 +38,34 @@ func NewClient(apiKey, team string) *Client {
 		apiKey,
 		team,
 		"https://amara.org/api",
+		RateLimitProtection{
+			MinRetryDuration: 5 * time.Second,
+			MaxRetryDuration: 20 * time.Minute,
+		},
 		httpClient,
 	}
 }
 
+func (c *Client) EnableRateLimitProtection() {
+	c.rateLimitProtection.enabled = true
+	c.rateLimitProtection.triggered = false
+	c.rateLimitProtection.counter = 0
+}
+
+func (c *Client) DisableRateLimitProtection() {
+	c.rateLimitProtection.enabled = false
+	c.rateLimitProtection.triggered = false
+}
+
+func (c *Client) SetRateLimitProtection(rlp RateLimitProtection) {
+	c.rateLimitProtection = rlp
+}
+
 func (c *Client) doRequest(method, url string, body io.Reader) ([]byte, error) {
+	if c.rateLimitProtection.triggered {
+		return nil, fmt.Errorf("Amara API is currently being rate limited. try again later")
```

Comment:
if this is newsroom user facing, maybe something less technical like:
"Amara is currently busy. Please wait and try again later."
or
"Amara is currently receiving too many requests. Please wait and try again later."
possibly with 
"wait 10 minutes" or some such number

Reply:
ok, so this is not user facing. never mind!